uxarray/grid.py

```diff
@@ -372,6 +373,79 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
 
         return integral
 
+    # A one-time-used helper function that will only be used in the build_face_edges_connectivity map() function usage
+    def __replace_fill_val(self, row, index, rep_val):
```

Comment:
I thought about that, But it needs to access the grid.mesh2_face_nodes. I feel like passing the entire copy of grid.mesh2_face_nodes is not a very good idea here

I personally don't like creating a one-time-used function as well. However, since we need to avoid looping, we need to use the python. `map()` function in this case. (I couldn't figure out any better python vectorization at this moment), and to use a `map()` function in this specific scenario, I have to create a function for it. (I also tried lambda function, but lambda doesn't support this value replacement)

Please let me know if you have any better suggestion for this.

Reply:
> Instead of being a grid.py function, it looks like a good candidate as an internal function for helprs.py?

I thought about that, But it needs to access the grid.mesh2_face_nodes. I feel like passing the entire copy of grid.mesh2_face_nodes is not a very good idea here

I personally don't like creating a one-time-used function as well. However, since we need to avoid looping, we need to use the python. `map()` function in this case. (I couldn't figure out any better python vectorization at this moment), and to use a `map()` function in this specific scenario, I have to create a function for it. (I also tried lambda function, but lambda doesn't support this value replacement)

Please let me know if you have any better suggestion for this.

Reply:
Ok, that makes sense, thanks!

Reply:
One approach could be to store these one-off utilities in a separate helper file i.e. `_build_face_edges_helpers.py` or `_grid_helpers.py` and simply exclude it from our API. This should hide these functions enough to prevent any accidents described in your second bullet. As for your first bullet, I think having them stored in a separate helper file as @erogluorhan mentioned would be best.

For now though, it seems like this function isn't needed anymore after some of the changes you've made, but I think as our codebase grows, we'll need to keep organizational issues like this in mind.

Reply:
You could also define this helper function as an [inner function](https://realpython.com/inner-functions-what-are-they-good-for/)